### PR TITLE
Assorted calendar fixes

### PR DESCRIPTION
**Pull request by yankees714**
A few issues uncovered after comparing the exchange calendars against a third party source.

(See commits for details).

### CODE REVIEW

File: trading_calendars/exchange_calendar_xfra.py [hunk 1]
```diff
@@ -127,6 +128,16 @@ def regular_holidays(self):
...
         ])
 
+    @property
+    def adhoc_holidays(self):
+        return [
+            # Whit Monday observed in 2007, before it became regularly
+            # observed starting in 2015.
+            Timestamp('2007-05-28', tz='UTC'),
+            # Reformation Day was a German national holiday in 2017.
+            Timestamp('2017-10-31', tz='UTC'),
```
Comment (vikram-narayan) [inline trading_calendars/exchange_calendar_xfra.py]: For consistency with other adhocs, could define these as `WhitMonday2007`, `ReformationDay500thAnniversary` (or something to that effect) below the regular holidays in this file, and reference them here?

File: tests/resources/xfra.csv [hunk 2]
```diff
@@ -7541,3 +7544,23 @@
 2019-09-03 00:00:00+00:00,2019-09-03 07:01:00,2019-09-03 15:30:00
 2019-09-04 00:00:00+00:00,2019-09-04 07:01:00,2019-09-04 15:30:00
 2019-09-05 00:00:00+00:00,2019-09-05 07:01:00,2019-09-05 15:30:00
+2019-09-06 00:00:00+00:00,2019-09-06 07:01:00,2019-09-06 15:30:00
```
Comment (vikram-narayan) [inline tests/resources/xfra.csv]: what is causing these 2019 additions to each calendar's csv?

**Comment by vikram-narayan** [comment]
looks like there's also a flake8 error

File: tests/resources/xfra.csv (hunk 2 above)
Comment (yankees714) [reply tests/resources/xfra.csv]: I think the calendars go a year out, so when you regenerate this output it adds the days out to a year from now. I can remove these if we want.

File: trading_calendars/exchange_calendar_xfra.py (hunk 1 above)
Comment (yankees714) [reply trading_calendars/exchange_calendar_xfra.py]: Sure, I might also add `AdHoc` to the end of those names to differentiate them from the holiday calendars

**Pull request merged**